spi/src/main/java/org/mapstruct/spi/AccessorNamingStrategy.java

```
@@ -0,0 +1,134 @@
+     *
+     * @param methodName The method name.
+     * @return {@code true} if the method name can be a non-boolean getter, {@code false} otherwise.
+     */
+    boolean isNonBooleanGetterName(String methodName);
+
+    /**
+     * Determine if a method name defines a getter with a boolean return type.
+     * <p>
+     * Default implementation: Method name starts with "is" and is longer than 2 characters.
+     *
+     * @param methodName The method name.
+     * @return {@code true} if the method name can be a boolean getter, {@code false} otherwise.
+     */
+    boolean isBooleanGetterName(String methodName);
+
+    /**
+     * Determine if a method name defines a setter.
+     * <p>
+     * Default implementation: Method name starts with "set" and is longer than 3 characters.
+     *
+     * @param methodName The method name.
+     * @return {@code true} if the method name can be a setter, {@code false} otherwise.
+     */
+    boolean isSetterName(String methodName);
+
+    /**
+     * Determine if a method name defines an adder.
+     * <p>
+     * Default implementation: Method name starts with "add" and is longer than 3 characters.
+     *
+     * @param methodName The method name.
+     * @return {@code true} if the method name can be an adder, {@code false} otherwise.
+     */
+    boolean isAdderName(String methodName);
+
+    /**
+     * Extract the property name from a method name for which {@link #isNonBooleanGetterName(String)} returned
+     * {@code true}.
+     * <p>
+     * Default implementation: Remove first 3 characters ("get") of the method name, and
+     * {@link java.beans.Introspector#decapitalize(String) decapitalize} the result.
+     *
+     * @param methodName The method name, guaranteed to be a non-boolean getter name.
+     * @return The property name corresponding to the method name.
+     */
+    String getPropertyNameForNonBooleanGetterName(String methodName);
```

You're right, for those three it should be perfectly fine. I'd like to keep the interface methods on the same level of abstraction, so if we pass one of them an `ExecutableElement` instead of a `String`, all of them should get an `ExecutableElement`. The `getPropertyName()` method might need to throw an exception if no property name can be determined - that currently happens in the `Executables` class.